kitsune/questions/utils.py

```diff
@@ -186,20 +189,28 @@ def process_classification_result(
                 reason=FlaggedObject.REASON_SPAM,
             )
         case _:
-            if topic_title := result["topic_result"].get("topic"):
-                try:
-                    topic = Topic.active.get(title=topic_title, visible=True)
-                except (Topic.DoesNotExist, Topic.MultipleObjectsReturned):
-                    return
-                else:
-                    flag_question(
-                        question,
-                        by_user=sumo_bot,
-                        notes=(
-                            "LLM classified as {topic.title}, for the following reason:\n"
-                            f"{result['topic_result']['reason']}"
-                        ),
-                        status=FlaggedObject.FLAG_ACCEPTED,
-                    )
+            if not (topic_title := result["topic_result"].get("topic")):
+                return
+
+            try:
+                topic = Topic.active.get(title=topic_title, visible=True)
+            except (Topic.DoesNotExist, Topic.MultipleObjectsReturned):
+                return
+
+            with transaction.atomic():
```

Comment:
Why is this needed?

Reply:
I removed it. As longs as the `question_classifier` task is made idempotent, the task retry mechanism should take care of ensuring synchronization if any one of the steps fails.